src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -1610,7 +1610,30 @@ private function skip_script_data(): bool {
 			 */
 			$at += 6;
 			$c   = $html[ $at ];
-			if ( ' ' !== $c && "\t" !== $c && "\r" !== $c && "\n" !== $c && '/' !== $c && '>' !== $c ) {
+			if (
+			/*
+			 * These characters trigger state transitions of interest:
+			 *
+			 * - @see {https://html.spec.whatwg.org/multipage/parsing.html#script-data-end-tag-name-state}
+			 * - @see {https://html.spec.whatwg.org/multipage/parsing.html#script-data-escaped-end-tag-name-state}
+			 * - @see {https://html.spec.whatwg.org/multipage/parsing.html#script-data-double-escape-start-state}
+			 * - @see {https://html.spec.whatwg.org/multipage/parsing.html#script-data-double-escape-end-state}
+			 *
+			 * The "\r" character is not present in the above references. However, "\r" must be
+			 * treated the same as "\n". This is because the HTML Standard requires newline
+			 * normalization during preprocessing which applies this replacement.
+			 *
+			 * - @see https://html.spec.whatwg.org/multipage/parsing.html#preprocessing-the-input-stream
+			 * - @see https://infra.spec.whatwg.org/#normalize-newlines
+			 */
+				'>' !== $c &&
+				' ' !== $c &&
+				"\n" !== $c &&
+				'/' !== $c &&
+				"\t" !== $c &&
+				"\f" !== $c &&
+				"\r" !== $c
```

Comment:
I ordered these by how often I'd expect the character to be seen in this position. I don't expect any real performance improvements from that part of the change, but also don't see any down side to having `>` and ` ` appear as the first and second match opportunities.

Reply:
my understanding is that these are all likely to be performed in parallel and executed before the CPU even reaches these lines, so yeah, I would _guess_ this is true, but without measurement also lean on not knowing. shouldn’t matter in any case, and unless someone has realistic benchmarks on realistic data, I would be skeptical of any performance claims on the position of these items.